contracts/0.8.25/vaults/VaultFactory.sol

```diff
@@ -84,8 +90,13 @@ contract VaultFactory {
             _stakingVaultInitializerExtraParams
         );
 
+        vault.fund{value: msg.value}();
+        vault.lock(connectDeposit);
```

Comment:
needed for auto-connect? but how it'll work on re-connected vaults where lock is pre-required?